### PR TITLE
Improvements to mustrunas_test.go

### LINKED ISSUE #41990: pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go could be improved
I'm suggesting to do the following improvements to the [pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go](https://github.com/kubernetes/kubernetes/blob/17375fc59fff39135af63bd1750bb07c36ef873b/pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go) file:

- `TestGenerateAdds`: remove unused `requiredDropCaps` member from the struct
- `TestValidateAdds`: remove unused `requiredDropCaps` member from the struct
- `TestValidateDrops`: remove unused `defaultAddCaps` member from the struct
- `TestValidateAdds`, `TestValidateDrops`: instead of using `shouldPass bool` and checking only whether test fails or not, we could check (also) an error message

This refactoring is simple and may be performed by the new contributors (could someone add label `for-new-contributors`?)

### PR DESCRIPTION
**Pull request by roidelapluie**
refs #41990

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #41990

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by pweil-** [comment]
@php-coder

**Comment by php-coder** [comment]
Wow, thank you @roidelapluie ! First 3 points now addressed, are you planning to fix also the 4th point?

BTW it's better to not specify issue number in the commit message because commits could be modified many times before being merged and all they will appear in the issue.

**Comment by roidelapluie** [comment]
Yes it is coming

**Comment by roidelapluie** [comment]
Here you go

**Comment by roidelapluie** [comment]
Ready for review

**Comment by roidelapluie** [comment]
Needs release-note-label-none

File: pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go [hunk 1]
```diff
@@ -216,23 +214,19 @@ func TestGenerateDrops(t *testing.T) {
...
-		},
+		"no required, no allowed, no container requests": {},
 		"no required, allowed, no container requests": {
 			allowedCaps: []api.Capability{"foo"},
-			shouldPass:  true,
 		},
 		"required, no allowed, no container requests": {
 			defaultAddCaps: []api.Capability{"foo"},
-			shouldPass:     false,
+			expectedErrors: []error{errors.New("capabilities: Invalid value: \"null\": required capabilities are not set on the securityContext")},
```
Comment (php-coder) [inline pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go]: WDYT about using back quotes? In this case we don't need to escape double quotes.

File: pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go [hunk 2]
```diff
@@ -308,36 +298,35 @@ func TestValidateAdds(t *testing.T) {
...
 		errs := strategy.Validate(nil, container)
-		if v.shouldPass && len(errs) > 0 {
-			t.Errorf("%s should have passed but had errors %v", k, errs)
+		if len(v.expectedErrors) != len(errs) {
+			t.Errorf("Expected %v (%v) errors, got %v (%v)", len(v.expectedErrors), v.expectedErrors, len(errs), errs)
 			continue
 		}
-		if !v.shouldPass && len(errs) == 0 {
-			t.Errorf("%s should have failed but received no errors", k)
+		for i := 0; i < len(v.expectedErrors); i++ {
```
Comment (php-coder) [inline pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go]: As far I can see there is always only one error. Maybe make it less general and test only for a single error?

Comment (roidelapluie) [inline pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go]: - Made it less general
- Then I also stopped creating errors and just came back to strings

File: pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go (hunk 1 above)
Comment (roidelapluie) [inline pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go]: Fixed

File: pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go (hunk 2 above)
Comment (php-coder) [inline pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go]: Thanks! Now tests are more readable.

**Comment by roidelapluie** [comment]
Hi @lavalamp what about this? Can we move forward?

cc @thockin

**Comment by php-coder** [comment]
Maybe @smarterclayton will find a second to approve this. :)

**Comment by smarterclayton** [comment]
This looks good but isn't required for 1.6 and we're in code freeze.  This looks good to merge when we open for 1.7 though.

**Comment by php-coder** [comment]
@smarterclayton Now, when 1.6 was released, could this be merged?

**Comment by deads2k** [comment]
@k8s-bot gci gce e2e test this

**Pull request merged**